pkg/daemon/controller/controller.go

```diff
@@ -153,6 +153,30 @@ func (sc *ServiceController) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(sc)
 }
 
+// RunBootstrap starts all Windows services that are pre-requisites to create a node object as defined in the given data
+func RunBootstrap(encodedCMData string) error {
+	cmData, err := servicescm.DecodeAndUnmarshall(encodedCMData)
+	if err != nil {
+		return errors.Wrap(err, "unable to decode ConfigMap data")
+	}
+
+	svcMgr, err := winsvc.NewMgr()
+	if err != nil {
+		return err
+	}
+	serviceController := NewServiceController(ctrl.SetupSignalHandler(), nil, svcMgr, "")
+
+	bootstrapSvcs := []servicescm.Service{}
+	for _, svc := range cmData.Services {
+		if !svc.Bootstrap {
+			// services are pre-sorted by priority, with all bootstrap services ordered towards the front of the slice
+			break
+		}
+		bootstrapSvcs = append(bootstrapSvcs, svc)
+	}
+	return serviceController.reconcileServices(bootstrapSvcs)
```

Comment:
Open question: since `RunBootstrap()` is majorly functions a wrapper for `reconcileServices()` do the existing reconcileService unit tests suffice?

Reply:
Yes if the 'wrapper' doesn't have any real logic.
The presence of this:
```
	cmData, err := servicescm.DecodeAndUnmarshall(encodedCMData)
	if err != nil {
		return errors.Wrap(err, "unable to decode ConfigMap data")
	}

	bootstrapSvcs := []servicescm.Service{}
	for _, svc := range cmData.Services {
		if !svc.Bootstrap {
			// services are pre-sorted by priority, with all bootstrap services ordered towards the front of the slice
			break
		}
		bootstrapSvcs = append(bootstrapSvcs, svc)
	}
```
makes it less of a wrapper, and i'm somewhat inclined to think this needs to be unit tested.

I'm back and forth on wondering if the signature of `reconcileServices` should be changed to accept a `bootstrap bool`, so that  that way there's less stuff going on in this wrapper.

Reply:
What I'll do is make the new signature `RunBootstrap(svcMgr winsvc.Mgr, encodedCMData string)` to make it unit testable, and leave the bootstrap logic here. `reconcileServices` does enough on it's own as a single func